code/modules/mining/equipment/monster_organs/monster_organ.dm

```diff
@@ -0,0 +1,202 @@
+	/// Icon state to apply when inert
+	var/icon_state_inert
+	/// Description to use once organ is inert
+	var/desc_inert
+	/// Status effect applied by this organ
+	var/datum/status_effect/user_status
+	/// Moodlet applied by this organ
+	var/datum/mood_event/moodlet
+	/// Action to grant when organ is implanted
+	var/datum/action/item_action/organ_action/use_internal
+
+/obj/item/organ/internal/monster_core/Initialize(mapload)
+	. = ..()
+	decay_timer = addtimer(CALLBACK(src, .proc/go_inert), time_to_decay, TIMER_STOPPABLE)
+	setup_internal_use_action()
+	add_item_action(use_internal)
+
+/// Set up the action for using the organ internally
+/obj/item/organ/internal/monster_core/proc/setup_internal_use_action()
+	use_internal = new(src)
+
+/obj/item/organ/internal/monster_core/Destroy(force, silent)
+	QDEL_NULL(use_internal)
+	return ..()
+
+/obj/item/organ/internal/monster_core/Insert(mob/living/carbon/target_carbon, special = 0, drop_if_replaced = TRUE)
+	. = ..()
+	if (inert)
+		to_chat(owner, span_notice("[src] breaks down as you try to insert it."))
+		qdel(src)
+		return FALSE
+	if (!decay_timer)
+		return TRUE
+	preserve(TRUE)
+	owner.visible_message(span_notice("[src] stabilizes as it's inserted."))
+	return TRUE
+
+/obj/item/organ/internal/monster_core/Remove(mob/living/carbon/target_carbon, special = 0)
+	if (!inert && !special)
+		owner.visible_message(span_notice("[src] rapidly decays as it's removed."))
+		go_inert()
+	return ..()
+
+/**
+ * Preserves the organ so that it will not decay.
+ * Returns true if successful.
+ * * Implanted - If true, organ has just been inserted into someone.
+ */
+/obj/item/organ/internal/monster_core/proc/preserve(implanted = FALSE)
+	if (inert)
+		return FALSE
+	deltimer(decay_timer)
+	decay_timer = null
+	update_appearance()
+	return TRUE
+
+/**
+ * Decays the organ, it is now useless.
+ */
+/obj/item/organ/internal/monster_core/proc/go_inert()
+	if (inert)
+		return FALSE
+	inert = TRUE
+	decay_timer = null
+	name = "decayed [name]"
+	update_appearance()
+	return TRUE
+
+/obj/item/organ/internal/monster_core/update_desc()
+	if (inert)
+		desc = (desc_inert) ? desc_inert : initial(desc)
+		return ..()
+	if (!decay_timer)
+		desc = (desc_preserved) ? desc_preserved : initial(desc)
+		return ..()
+	desc = initial(desc)
+	return ..()
+
+/obj/item/organ/internal/monster_core/update_icon_state()
+	if (inert)
+		icon_state = (icon_state_inert) ? icon_state_inert : initial(icon_state)
+		return ..()
+	if (!decay_timer)
+		icon_state = (icon_state_preserved) ? icon_state_preserved : initial(icon_state)
+		return ..()
+	icon_state = initial(icon_state)
+	return ..()
+
+/obj/item/organ/internal/monster_core/afterattack(atom/target, mob/user, proximity_flag)
+	. = ..()
+	if (!proximity_flag)
+		return
+	try_apply(target, user)
+
+/obj/item/organ/internal/monster_core/attack_self(mob/user)
+	if (!user.canUseTopic(src, be_close = TRUE, no_dexterity = FALSE, no_tk = TRUE))
+		return
+	try_apply(user, user)
```

Comment:
you know, at some point you gotta ask, why is this even an organ.

Reply:
You're not really wrong that probably the instances of people implanting even the existing item as an organ happens one round in a hundred (even though as you noticed, it provides a full admin heal!) as opposed to just using it inhand as soon as you see it on the floor
I think that for the sandbox it's still fun to have the option to see a weird alien organ and have the ability to ask a doctor to put it into your body though